### PR TITLE
Add support for ablation evaluation

### PR DESCRIPTION
**Pull request by rbiswasfc**
This PR adds script and configs to run ablation evaluations with 4 datasets (MNLI, WiC, BoolQ, EurLEX). BoolQ and WiC fine-tuning uses checkpoint obtained from the MNLI training run. 

Eval for flex bert can be run using `python ablation_eval.py yamls/ablations/flex-bert-ablation-eval.yaml`, which gives following scores:
```
-----------------------------------------------------------------------------------------------------------------
Job                                               | Dataset                  | Metric                     |
-----------------------------------------------------------------------------------------------------------------
MNLI(seed=23)                                     | glue_mnli_mismatched     | MulticlassAccuracy         | 84.53
EURLEX(seed=23)                                   | long_context_eurlex      | EurlexMultilabelF1Score    | 71.85
BOOLQ(seed=23)                                    | superglue_boolq          | MulticlassAccuracy         | 78.35
WIC(seed=23)                                      | superglue_wic            | MulticlassAccuracy         | 65.99
-----------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
This looks good. Can you add a smoketest for `ablation_eval.py` like we have for `glue.py`? Then it will be ready to merge.

**Comment by rbiswasfc** [comment]
> This looks good. Can you add a smoketest for `ablation_eval.py` like we have for `glue.py`? Then it will be ready to merge.

Thanks for the suggestion, just added the smoketest!

**Review by bclavie — APPROVED** [review]
LGTM with the smoketest added. The results are _slightly_ lower than they ought to be in an ideal world, but it still looks in line and could just be a random seed/hyperparam thing. Not off by enough to be alarming IMO.

Thank you!

**Comment by NohTow** [comment]
LGTM, as raised, the results are still a bit low but this seems ok and this does not come from the PR as it was already the case with my prior tests.
